apps/src/applab/designElements/label.jsx

```diff
@@ -147,25 +147,30 @@ module.exports = {
   },
 
   resizeToFitText: function (element) {
-    var clone = $(element).clone().css({
-      position: 'absolute',
-      visibility: 'hidden',
-      width: 'auto',
-      height: 'auto',
-      maxWidth: (Applab.appWidth - parseInt(element.style.left, 10)) + 'px',
-    }).appendTo($(document.body));
-
-    var padding = parseInt(element.style.padding, 10);
-
-    if ($(element).data('lock-width') !== PropertyRow.LockState.LOCKED) {
-      //Truncate the width before it runs off the edge of the screen
-      element.style.width = Math.min(clone.width() + 1 + 2 * padding, Applab.appWidth - clone.position().left) + 'px';
+    if (element.textContent) {
+      var clone = $(element).clone().css({
+        position: 'absolute',
+        visibility: 'hidden',
+        width: 'auto',
+        height: 'auto',
+        maxWidth: (Applab.appWidth - parseInt(element.style.left, 10)) + 'px',
+      }).appendTo($(document.body));
+
+      var padding = parseInt(element.style.padding, 10);
+
+      if ($(element).data('lock-width') !== PropertyRow.LockState.LOCKED) {
+        //Truncate the width before it runs off the edge of the screen
+        element.style.width = Math.min(clone.width() + 1 + 2 * padding, Applab.appWidth - clone.position().left) + 'px';
+      }
+      if ($(element).data('lock-height') !== PropertyRow.LockState.LOCKED) {
+        element.style.height = clone.height() + 1 + 2 * padding + 'px';
+      }
+
+      clone.remove();
+    } else {
+      element.style.width = '15px';
+      element.style.height = '15px';
```

Comment:
Maybe a short comment explaining that when we have no text and no width/height specified, we default to 15/15